src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -9,6 +9,144 @@
+		protected virtual void OnAdded(T view)
+		{
+		}
+
+		/// <summary>
+		/// Invoked when a child is removed the layout. Implement this method to add class handling for this event.
+		/// </summary>
+		/// <param name="view">The view which was removed.</param>
+		protected virtual void OnRemoved(T view)
+		{
+		}
+
+		[Obsolete("Use ZIndex")]
+		public void LowerChild(View view)
+		{
+			if (!this.Contains(view) || this.First() == view)
+			{
+				return;
+			}
+
+			this.Remove(view);
+			this.Insert(view, 0);
+			OnChildrenReordered();
+		}
+
+		[Obsolete("Use ZIndex")]
+		public void RaiseChild(View view)
+		{
+			if (!this.Contains(view) || this.Last() == view)
+			{
+				return;
+			}
+
+			this.Remove(view);
+			this.Insert(view,this.Count - 1);
+			OnChildrenReordered();
+		}
+
+		/// <summary>
+		/// Instructs the layout to relayout all of its children.
+		/// </summary>
+		/// <remarks>This method starts a new layout cycle for the layout. Invoking this method frequently can negatively impact performance.</remarks>
+		protected virtual void InvalidateLayout()
+		{
+			// Todo call this when InvalidateMeasure is called?
+			(this as IView).InvalidateMeasure();
+		}
+
+		Graphics.Size _layoutChildren = Graphics.Size.Zero;
+		Graphics.Size ICrossPlatformLayout.CrossPlatformArrange(Graphics.Rect bounds)
+		{
+			_layoutChildren = bounds.Size;
+#pragma warning disable CS0618 // Type or member is obsolete
+			LayoutChildren(bounds.X, bounds.Y, bounds.Width, bounds.Height);
+#pragma warning restore CS0618 // Type or member is obsolete
+			return _layoutChildren;
+		}
+
+
+		[Obsolete("Use your own LayoutManager")]
+		protected virtual void LayoutChildren(double x, double y, double width, double height)
+		{
+			_layoutChildren = base._layoutManager.ArrangeChildren(new Graphics.Rect(x, y, width, height));
+		}
+
+		[Obsolete("This method on the original class makes no sense and was probably accidently made protected. I'll write a better message before merging.")]
+		protected void OnChildMeasureInvalidated(object sender, EventArgs e)
+		{
+			OnChildMeasureInvalidated();
+		}
+
+		protected virtual void OnChildMeasureInvalidated()
+		{
+		}
+
+		
+		[Obsolete("Obsolete. This never really did anything because the platform is going to invalidate the layout anyway when views are removed/added.")]
+		protected virtual bool ShouldInvalidateOnChildAdded(View child) => true;
+		
+		[Obsolete("Obsolete. This never really did anything because the platform is going to invalidate the layout anyway when views are removed/added.")]
+		protected virtual bool ShouldInvalidateOnChildRemoved(View child) => true;
+		
+		/// <summary>
+		/// Instructs the layout to relayout all of its children.
+		/// </summary>
+		/// <remarks>This method starts a new layout cycle for the layout. Invoking this method frequently can negatively impact performance.</remarks>
+		protected void UpdateChildrenLayout()
+		{
+			(this as IView).InvalidateMeasure();
```

Comment:
This does the same as `InvalidateLayout` above, so should this also be obsolete?